src/utility.h

```diff
@@ -0,0 +1,331 @@
+#ifndef CHKSTAT_UTILITY_H
+#define CHKSTAT_UTILITY_H
+
+// POSIX
+#include <sys/capability.h>
+#include <sys/stat.h>
+
+// third party
+#include <tclap/CmdLine.h>
+
+// C++
+#include <cctype>
+#include <cstring>
+#include <string>
+#include <string_view>
+#include <vector>
+
+/**
+ * \brief
+ *     SwitchArg that can be programmatically set
+ * \details
+ *     TCLAP::SwitchArg doesn't offer a public API to programmatically change
+ *     the switch's value. Therefore this specializations provides an
+ *     additional method to make this possible.
+ **/
+class SwitchArgRW :
+    public TCLAP::SwitchArg
+{
+public:
+    SwitchArgRW(
+        const std::string &flag,
+        const std::string &name,
+        const std::string &desc,
+        TCLAP::CmdLineInterface &parser) :
+        TCLAP::SwitchArg(flag, name, desc, parser)
+    {}
+
+    void setValue(bool val)
+    {
+        _value = val;
+        // this is used for isSet(), _value only for getValue(), so
+        // sync both.
+        _alreadySet = val;
+    }
+};
+
+// isspace has overloads which gives trouble with template argument deduction,
+// therefore provide a wrapper
+inline bool chkspace(char c) { return std::isspace(c); }
+
+//! remove certain leading characters from the given string (by default
+//! whitespace characters)
+template <typename UNARY = bool(char)>
+inline std::string& lstrip(std::string &s, UNARY f = chkspace)
```

Comment:
I had more `std::string_view` in there while working on it but somehow it became difficult having them in interfaces, keeping it consistent, since `string_view` cannot be used interchangeable with `string`.

Yeah I can make them return nothing. Often I introduce sibbling functions like `stripped()` that return a modified copy instead. But I think those variants aren't needed currently in the code.